src/software/new_geom/cubic_bezier_spline2d_test.cpp

```diff
@@ -0,0 +1,235 @@
+#include "software/new_geom/cubic_bezier_spline2d.h"
+
+#include <gtest/gtest.h>
+
+#include "software/new_geom/bezier_curve2d.h"
+#include "software/test_util/test_util.h"
+
+class CubicBezierSplineTest : public ::testing::Test
+{
+   protected:
+    CubicBezierSplineTest()
+        : test_spline_1(Point(1, -1), Vector(-3, -4), Point(1, -2), Vector(-2, -5),
+                        {
+                            Point(5, 5),
+                            Point(10, -10),
+                        })
+    {
+    }
+
+    void SetUp() override {}
+
+    void TearDown() override {}
+
+    // In order to avoid duplicate work between tests we create splines here that are
+    // used across many tests
+    const CubicBezierSpline2d test_spline_1;
+};
+
+TEST_F(CubicBezierSplineTest, get_num_knots)
+{
+    // Two intermediate knots + start point + end point
+    EXPECT_EQ(4, test_spline_1.getNumKnots());
+}
+
+TEST_F(CubicBezierSplineTest, get_knots)
+{
+    const std::vector<Point> expected_knots = {Point(1, -1), Point(5, 5), Point(10, -10),
+                                               Point(1, -2)};
+    EXPECT_EQ(expected_knots, test_spline_1.getKnots());
+}
+
+TEST_F(CubicBezierSplineTest, get_knot_parametrization_values)
+{
+    const std::vector<Point> expected_knots = {Point(1, -1), Point(5, 5), Point(10, -10),
+                                               Point(1, -2)};
+    const std::vector<double> knot_vector = test_spline_1.getKnotParametrizationValues();
+
+    ASSERT_EQ(4, knot_vector.size());
+
+    for (size_t i = 0; i < knot_vector.size(); i++)
+    {
+        EXPECT_EQ(test_spline_1.getValueAt(knot_vector[i]), expected_knots[i]);
+    }
+    EXPECT_EQ(expected_knots, test_spline_1.getKnots());
+}
+
+TEST_F(CubicBezierSplineTest, getValueAt__start_point)
```

Comment:
Are the 2 underscores intentional here?

Reply:
yeah, they're separating the function we're testing from the test case name for that function